src/v2i-hub/CDASimAdapter/src/CDASimConnection.cpp

```diff
@@ -217,4 +224,67 @@ namespace CDASimAdapter{
         forward_message( msg , message_receiver_publisher );
     }
 
+    void CDASimConnection::populate_sensors_with_file(const std::string file_path, Json::Value& sensors_json_v){
+        //Read file from disk
+        std::ifstream in_strm;
+        in_strm.open(file_path, std::ifstream::binary);
+        if(!in_strm.is_open())
+        {
+            PLOG(logERROR) << "File cannot be opened. File path: " << file_path <<std::endl;
+            return;
+        }
+        std::string line;
+        std::stringstream ss;
+        while (std::getline(in_strm, line)) {
+            ss << line;
+        }
+        in_strm.close();
+
+       populate_json_with_string(ss.str(), sensors_json_v);
+    }
+
+    void CDASimConnection::populate_json_with_string(const std::string& json_str, Json::Value& json_v){
+        //Update JSON value with information from file
+        Json::CharReaderBuilder builder;
+        std::unique_ptr<Json::CharReader> reader(builder.newCharReader());
+        JSONCPP_STRING err;
+        if(!reader->parse(json_str.c_str(), json_str.c_str() + json_str.length(), &json_v, &err))
+        {
+            PLOG(logERROR) << "Error parsing sensors from string: " << json_str << std::endl;
+            return;
+        }
+    }
+
+    Json::Value CDASimConnection::get_sensor_by_id(std::string &sensor_id)
```

Comment:
Just out of curiosity, is there a reason you opted against making sensor its own data type that handles serialization/deserialization internally. This works to but it seems to be that it may be a cleaner implementation of we keep JSON serialization/deserialization and data type definition separate from this plugin.

Reply:
Do we need to create another data type for sensor in JSON representation? This function is to find the sensor Json::Value from the sensors of type Json::Value which is sent for registration using sensor id. As we will need  this later to modify the detected object.

Reply:
I think Json::Value data type is sufficient for this case. It allows more flexibility to modify the sensor JSON, and no overhead defining another strong sensor object type, then implement our own serialization/deserialization logic.

Reply:
I think we are back to our definition of detected object. As we forwarding detected object as JSON string to TMX bus, and subscribe the JSON string as it. I do not think we need strong type and operation here.